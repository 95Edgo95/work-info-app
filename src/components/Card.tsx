import CardRow from "components/CardRow";
import Link from "react-router-dom/Link";
import * as React from "react";

import * as styles from "assets/css/card.css";

interface ICardProps {
  title?: string;
  link?: string;
  details: Array<{
    tooltip: string;
    icon: string;
    value: any;
  }>;
  children: any;
}

export default function Card({ title = '', link = '', details, children }: ICardProps) {
  const content = (
    <React.Fragment>
      {children}
      {title && (
        <div className={styles.detail}>
          {title}
        </div>
      )}
      {details.map(({value, icon, tooltip}) => (
        <CardRow
          tooltip={tooltip}
          value={value}
          icon={icon}
          key={value}
        />
      ))}
    </React.Fragment>
  );

  return link ? (
    <Link
      to={link}
      className={styles.card}
    >
      {content}
    </Link>
  ) : (
    <div className={styles.card}>
      {content}
    </div>
  );
}
