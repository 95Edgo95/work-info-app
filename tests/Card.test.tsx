import Card from "../src/components/Card";
import { mount } from "enzyme";
import * as React from "react";
import {BrowserRouter} from "react-router-dom";

describe('Card', () => {
  it('should render correctly', () => {
    const component = mount(
      <Card
        details={[
          {
            value: "test@test.com",
            icon: "email",
          },
          {
            value: "TE12345678",
            icon: "passport",
          },
          {
            value: "14/11/1995",
            icon: "birthday",
          },
          {
            value: 5,
            icon: "workplace",
          },
        ]}
      >
        <div />
      </Card>
    );

    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it('should render correctly with title', () => {
    const component = mount(
      <Card
        details={[
          {
            value: "test@test.com",
            icon: "email",
          },
          {
            value: "TE12345678",
            icon: "passport",
          },
          {
            value: "14/11/1995",
            icon: "birthday",
          },
          {
            value: 5,
            icon: "workplace",
          },
        ]}
        title="Title"
      >
        <div />
      </Card>
    );

    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it('should render correctly with link', () => {
    const component = mount(
      <BrowserRouter>
        <Card
          details={[
            {
              value: "test@test.com",
              icon: "email",
            },
            {
              value: "TE12345678",
              icon: "passport",
            },
            {
              value: "14/11/1995",
              icon: "birthday",
            },
            {
              value: 5,
              icon: "workplace",
            },
          ]}
          link="/test/id"
        >
          <div />
        </Card>
      </BrowserRouter>
    );

    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it('should render correctly with title and link', () => {
    const component = mount(
      <BrowserRouter>
        <Card
          details={[
            {
              value: "test@test.com",
              icon: "email",
            },
            {
              value: "TE12345678",
              icon: "passport",
            },
            {
              value: "14/11/1995",
              icon: "birthday",
            },
            {
              value: 5,
              icon: "workplace",
            },
          ]}
          title="Title"
          link="/test/id"
        >
          <div />
        </Card>
      </BrowserRouter>
    );

    expect(component).toMatchSnapshot();
    component.unmount();
  });
});
