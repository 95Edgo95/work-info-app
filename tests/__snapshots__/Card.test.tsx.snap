// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<Card
  details={
    Array [
      Object {
        "icon": "email",
        "value": "test@test.com",
      },
      Object {
        "icon": "passport",
        "value": "TE12345678",
      },
      Object {
        "icon": "birthday",
        "value": "14/11/1995",
      },
      Object {
        "icon": "workplace",
        "value": 5,
      },
    ]
  }
>
  <div>
    <div />
    <CardRow
      icon="email"
      key="test@test.com"
      value="test@test.com"
    >
      <div>
        <Icon
          name="email"
        >
          <span
            className="undefined "
            role="presentation"
            style={
              Object {
                "color": "black",
                "fontSize": "24px",
              }
            }
          >
            <svg>
              <use
                xlinkHref="#email"
              />
            </svg>
          </span>
        </Icon>
        <span>
          test@test.com
        </span>
      </div>
    </CardRow>
    <CardRow
      icon="passport"
      key="TE12345678"
      value="TE12345678"
    >
      <div>
        <Icon
          name="passport"
        >
          <span
            className="undefined "
            role="presentation"
            style={
              Object {
                "color": "black",
                "fontSize": "24px",
              }
            }
          >
            <svg>
              <use
                xlinkHref="#passport"
              />
            </svg>
          </span>
        </Icon>
        <span>
          TE12345678
        </span>
      </div>
    </CardRow>
    <CardRow
      icon="birthday"
      key="14/11/1995"
      value="14/11/1995"
    >
      <div>
        <Icon
          name="birthday"
        >
          <span
            className="undefined "
            role="presentation"
            style={
              Object {
                "color": "black",
                "fontSize": "24px",
              }
            }
          >
            <svg>
              <use
                xlinkHref="#birthday"
              />
            </svg>
          </span>
        </Icon>
        <span>
          14/11/1995
        </span>
      </div>
    </CardRow>
    <CardRow
      icon="workplace"
      key="5"
      value={5}
    >
      <div>
        <Icon
          name="workplace"
        >
          <span
            className="undefined "
            role="presentation"
            style={
              Object {
                "color": "black",
                "fontSize": "24px",
              }
            }
          >
            <svg>
              <use
                xlinkHref="#workplace"
              />
            </svg>
          </span>
        </Icon>
        <span>
          5
        </span>
      </div>
    </CardRow>
  </div>
</Card>
`;

exports[`Card should render correctly with link 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Card
      details={
        Array [
          Object {
            "icon": "email",
            "value": "test@test.com",
          },
          Object {
            "icon": "passport",
            "value": "TE12345678",
          },
          Object {
            "icon": "birthday",
            "value": "14/11/1995",
          },
          Object {
            "icon": "workplace",
            "value": 5,
          },
        ]
      }
      link="/test/id"
    >
      <Link
        replace={false}
        to="/test/id"
      >
        <a
          href="/test/id"
          onClick={[Function]}
        >
          <div />
          <CardRow
            icon="email"
            key="test@test.com"
            value="test@test.com"
          >
            <div>
              <Icon
                name="email"
              >
                <span
                  className="undefined "
                  role="presentation"
                  style={
                    Object {
                      "color": "black",
                      "fontSize": "24px",
                    }
                  }
                >
                  <svg>
                    <use
                      xlinkHref="#email"
                    />
                  </svg>
                </span>
              </Icon>
              <span>
                test@test.com
              </span>
            </div>
          </CardRow>
          <CardRow
            icon="passport"
            key="TE12345678"
            value="TE12345678"
          >
            <div>
              <Icon
                name="passport"
              >
                <span
                  className="undefined "
                  role="presentation"
                  style={
                    Object {
                      "color": "black",
                      "fontSize": "24px",
                    }
                  }
                >
                  <svg>
                    <use
                      xlinkHref="#passport"
                    />
                  </svg>
                </span>
              </Icon>
              <span>
                TE12345678
              </span>
            </div>
          </CardRow>
          <CardRow
            icon="birthday"
            key="14/11/1995"
            value="14/11/1995"
          >
            <div>
              <Icon
                name="birthday"
              >
                <span
                  className="undefined "
                  role="presentation"
                  style={
                    Object {
                      "color": "black",
                      "fontSize": "24px",
                    }
                  }
                >
                  <svg>
                    <use
                      xlinkHref="#birthday"
                    />
                  </svg>
                </span>
              </Icon>
              <span>
                14/11/1995
              </span>
            </div>
          </CardRow>
          <CardRow
            icon="workplace"
            key="5"
            value={5}
          >
            <div>
              <Icon
                name="workplace"
              >
                <span
                  className="undefined "
                  role="presentation"
                  style={
                    Object {
                      "color": "black",
                      "fontSize": "24px",
                    }
                  }
                >
                  <svg>
                    <use
                      xlinkHref="#workplace"
                    />
                  </svg>
                </span>
              </Icon>
              <span>
                5
              </span>
            </div>
          </CardRow>
        </a>
      </Link>
    </Card>
  </Router>
</BrowserRouter>
`;

exports[`Card should render correctly with title 1`] = `
<Card
  details={
    Array [
      Object {
        "icon": "email",
        "value": "test@test.com",
      },
      Object {
        "icon": "passport",
        "value": "TE12345678",
      },
      Object {
        "icon": "birthday",
        "value": "14/11/1995",
      },
      Object {
        "icon": "workplace",
        "value": 5,
      },
    ]
  }
  title="Title"
>
  <div>
    <div />
    <div>
      Title
    </div>
    <CardRow
      icon="email"
      key="test@test.com"
      value="test@test.com"
    >
      <div>
        <Icon
          name="email"
        >
          <span
            className="undefined "
            role="presentation"
            style={
              Object {
                "color": "black",
                "fontSize": "24px",
              }
            }
          >
            <svg>
              <use
                xlinkHref="#email"
              />
            </svg>
          </span>
        </Icon>
        <span>
          test@test.com
        </span>
      </div>
    </CardRow>
    <CardRow
      icon="passport"
      key="TE12345678"
      value="TE12345678"
    >
      <div>
        <Icon
          name="passport"
        >
          <span
            className="undefined "
            role="presentation"
            style={
              Object {
                "color": "black",
                "fontSize": "24px",
              }
            }
          >
            <svg>
              <use
                xlinkHref="#passport"
              />
            </svg>
          </span>
        </Icon>
        <span>
          TE12345678
        </span>
      </div>
    </CardRow>
    <CardRow
      icon="birthday"
      key="14/11/1995"
      value="14/11/1995"
    >
      <div>
        <Icon
          name="birthday"
        >
          <span
            className="undefined "
            role="presentation"
            style={
              Object {
                "color": "black",
                "fontSize": "24px",
              }
            }
          >
            <svg>
              <use
                xlinkHref="#birthday"
              />
            </svg>
          </span>
        </Icon>
        <span>
          14/11/1995
        </span>
      </div>
    </CardRow>
    <CardRow
      icon="workplace"
      key="5"
      value={5}
    >
      <div>
        <Icon
          name="workplace"
        >
          <span
            className="undefined "
            role="presentation"
            style={
              Object {
                "color": "black",
                "fontSize": "24px",
              }
            }
          >
            <svg>
              <use
                xlinkHref="#workplace"
              />
            </svg>
          </span>
        </Icon>
        <span>
          5
        </span>
      </div>
    </CardRow>
  </div>
</Card>
`;

exports[`Card should render correctly with title and link 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Card
      details={
        Array [
          Object {
            "icon": "email",
            "value": "test@test.com",
          },
          Object {
            "icon": "passport",
            "value": "TE12345678",
          },
          Object {
            "icon": "birthday",
            "value": "14/11/1995",
          },
          Object {
            "icon": "workplace",
            "value": 5,
          },
        ]
      }
      link="/test/id"
      title="Title"
    >
      <Link
        replace={false}
        to="/test/id"
      >
        <a
          href="/test/id"
          onClick={[Function]}
        >
          <div />
          <div>
            Title
          </div>
          <CardRow
            icon="email"
            key="test@test.com"
            value="test@test.com"
          >
            <div>
              <Icon
                name="email"
              >
                <span
                  className="undefined "
                  role="presentation"
                  style={
                    Object {
                      "color": "black",
                      "fontSize": "24px",
                    }
                  }
                >
                  <svg>
                    <use
                      xlinkHref="#email"
                    />
                  </svg>
                </span>
              </Icon>
              <span>
                test@test.com
              </span>
            </div>
          </CardRow>
          <CardRow
            icon="passport"
            key="TE12345678"
            value="TE12345678"
          >
            <div>
              <Icon
                name="passport"
              >
                <span
                  className="undefined "
                  role="presentation"
                  style={
                    Object {
                      "color": "black",
                      "fontSize": "24px",
                    }
                  }
                >
                  <svg>
                    <use
                      xlinkHref="#passport"
                    />
                  </svg>
                </span>
              </Icon>
              <span>
                TE12345678
              </span>
            </div>
          </CardRow>
          <CardRow
            icon="birthday"
            key="14/11/1995"
            value="14/11/1995"
          >
            <div>
              <Icon
                name="birthday"
              >
                <span
                  className="undefined "
                  role="presentation"
                  style={
                    Object {
                      "color": "black",
                      "fontSize": "24px",
                    }
                  }
                >
                  <svg>
                    <use
                      xlinkHref="#birthday"
                    />
                  </svg>
                </span>
              </Icon>
              <span>
                14/11/1995
              </span>
            </div>
          </CardRow>
          <CardRow
            icon="workplace"
            key="5"
            value={5}
          >
            <div>
              <Icon
                name="workplace"
              >
                <span
                  className="undefined "
                  role="presentation"
                  style={
                    Object {
                      "color": "black",
                      "fontSize": "24px",
                    }
                  }
                >
                  <svg>
                    <use
                      xlinkHref="#workplace"
                    />
                  </svg>
                </span>
              </Icon>
              <span>
                5
              </span>
            </div>
          </CardRow>
        </a>
      </Link>
    </Card>
  </Router>
</BrowserRouter>
`;
